### ISSUE TITLE
2.0.0 - Indentation error on single-line case statement when allowing implicit indents

### ISSUE THREAD

**Issue by fredriv** [sqlfluff/sqlfluff#4542]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff v2.0.0 gives an LT02 indentation error on a single-line case statement within a multi-line select statement when `allow_implicit_indents = True`.

The same case statement does not produce linting errors if it is part of a single-line select statement with only one element selected.

### Expected Behaviour

I expected the SQL to parse without linting errors.

SQLFluff v1.4.5 does not report the same linting error.

### Observed Behaviour

```
L:   3 | P:  31 | LT02 | Expected line break and no indent before 'else'.
                       | [layout.indent]
L:   3 | P:  38 | LT02 | Expected line break and no indent before 'end'.
                       | [layout.indent]
```

### How to reproduce

```
-- Fails with an LT02 indentation error
select
    a,
    case when b is null then 0 else 1 end as c
from my_table;


-- No indentation error
select case when b is null then 0 else 1 end as c
from my_table;
```

### Dialect

`ansi`

### Version

```
> sqlfluff --version
sqlfluff, version 2.0.0

> python --version
Python 3.9.9
```

### Configuration

```
[sqlfluff]
dialect = ansi

[sqlfluff:indentation]
allow_implicit_indents = True
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Confirmed still a problem on `main`. I'll pick this one up.

**Comment by alanmcruickshank** [comment]
Thanks for putting the effort into such a clean test case @fredriv - that made this much easier to fix.

**Comment by martareda** [comment]
Hello, 
I have similar issue. What I have observed it happens when case statement is before from clause. 

```
select
    company_id
    , year_month
    , sum(people_revenue * charge_amount) as people_revenue
    , sum(premium_revenue * charge_amount) as premium_revenue
    , sum(expenses_revenue * charge_amount) as expenses_revenue
    , sum(case when product_rate_group = 'Without Add-on' then charge_amount else 0 end) as business_mrr_without_add_on
    , sum(case when product_rate_group != 'Without Add-on' then charge_amount else 0 end) as all_add_on_mrr
from companies_mrr_gspread_rate
group by 1, 2 
```

I have above code that fails only for the last column:

```
L:  38 | P:  78 | LT02 | Expected line break and no indent before 'else'.
                       | [layout.indent]
L:  38 | P:  85 | LT02 | Expected line break and no indent before 'end'.
                       | [layout.indent]
```


```
Configuration
[sqlfluff]
dialect = redshift

[sqlfluff:indentation]
allow_implicit_indents = True
```
